pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -441,7 +441,7 @@ func (p *PriorityQueue) isPodWorthRequeuing(logger klog.Logger, pInfo *framework
 	pod := pInfo.Pod
 	queueStrategy := queueSkip
 	for eventToMatch, hintfns := range hintMap {
-		if eventToMatch.Resource != event.Resource || eventToMatch.ActionType&event.ActionType == 0 {
+		if !eventToMatch.Match(event) {
```

Comment:
Does this code run if the feature gate is disabled?

That would decide whether we need to cherry-pick to older versions.

Reply:
Alternatively, we don't need to cherry pick if we don't use these kind of wildcards in the default scheduler. Do we?

Reply:
I don't think it's worth being cherry-picked.

We do reach here regardless of the feature gate though, no in-tree plugin uses this in `EventsToRegister`. Plus, I would say it was rare to use a wildcard event in `EventsToRegister` actually, since registering a wildcard equals leaving `EventsToRegister` empty. 
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/runtime/framework.go#L622-L626

Reply:
So, should we tag this like so?
````
/remove-kind bug
/kind cleanup
```

Reply:
I take it as a non-critical bug as https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks describes, make sense?

Reply:
SGTM

Reply:
+1